app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+        @tweets = Tweet.all
+        erb :'/tweets/tweets'
+    else
+        redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    #load create tweet form
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets/:id' do
+    #individual show tweet page
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    #load form to edit tweet
+    #find specific tweet from params[:id] to preload form
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/edit_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    #submit and catch the data and create new tweet object
+    #redirect to individual tweet page
+    if params[:content] == ""
+      redirect to "/tweets/new"
+    else
+    @tweet = Tweet.create(content: params[:content], user_id: session[:user_id])
+    @tweet.save
+    redirect "/tweets/#{@tweet.id}"
+    end
+  end
+
+  delete '/tweets/:id/delete' do
+    #delete tweet by :id
+    @tweet = Tweet.find_by_id(params[:id])
+    #binding.pry
+    if logged_in? && @tweet.user_id == current_user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.